File: 프로그래머스/lv2/42747. H－Index/H－Index.py
def solution(citations):
    answer = 0
    l = len(citations)
    citations=sorted(citations, reverse = True)
    h = citations[0]
    for i in range(h, 0, -1):
        len1 = len([x for x in citations if x<=i])
        len2 = len([x for x in citations if x>=i])
        if len1<=i and len2 >= i:
            return i
        
    
    return answer

File: 프로그래머스/lv2/42747. H－Index/H－Index.java
import java.util.*;
class Solution {
    public int solution(int[] citations) {
        int answer = 0;
        int len = citations.length;
        int max = 0;
        for(int i=len;i>0;i--){
            int cnt = 0;
            for(int c : citations){
                if(c>=i)
                    cnt++;
            }
            if(cnt>=i)
                max = (i>max) ? i : max;
        }
        return max;
    }
}